contrib/core-contract-tests/tests/pox-4/pox_StackAggregationCommitAuthCommand.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  logCommand,
+  PoxCommand,
+  Real,
+  Stub,
+  Wallet,
+} from "./pox_CommandModel.ts";
+import { poxAddressToTuple } from "@stacks/stacking";
+import { expect } from "vitest";
+import { Cl } from "@stacks/transactions";
+
+/**
+ * The `StackAggregationCommitAuthCommand` allows an operator commits partially
+ * stacked STX and allocate a new PoX reward address slot. This allows a
+ * stacker to lock fewer STX than the minimal threshold in multiple transactions,
+ * so long as:
+ *  1. The pox-addr is the same.
+ *  2. This "commit" transaction is called _before_ the PoX anchor block.
+ *
+ * This command calls stack-aggregation-commit using an `authorization`.
+ *
+ * Constraints for running this command include:
+ * - The Operator must have locked STX on behalf of at least one stacker.
+ * - The total amount previously locked by the Operator on behalf of the
+ *   stackers has to be greater than the uSTX threshold.
+ * - All of the Stackers must have delegated to the same pox address.
+ */
+export class StackAggregationCommitAuthCommand implements PoxCommand {
+  readonly operator: Wallet;
+  readonly authId: number;
+  readonly currentCycle: number;
+
+  /**
+   * Constructs a `StackAggregationCommitAuthCommand` to lock uSTX for stacking.
+   *
+   * @param operator - Represents the `Operator`'s wallet.
+   * @param authId - Unique `auth-id` for the authorization.
+   * @param currentCycle - The current reward cycle.
+   */
+  constructor(
+    operator: Wallet,
+    authId: number,
+    currentCycle: number,
+  ) {
+    this.operator = operator;
+    this.authId = authId;
+    this.currentCycle = currentCycle;
+  }
+
+  check(model: Readonly<Stub>): boolean {
+    // Constraints for running this command include:
+    // - The Operator must have locked STX on behalf of at least one stacker.
+    // - The total amount previously locked by the Operator on behalf of the
+    //   stackers has to be greater than the uSTX threshold.
+    // - All of the Stackers must have delegated to the same pox address.
+    let sameDelegatedPoxAddrAllStackers = true;
+    const firstDelegatedPoxAddress = this.operator.lockedAddresses.length > 0
+      ? model.wallets.get(this.operator.lockedAddresses[0])!.delegatedPoxAddress
+      : this.operator.btcAddress;
+
+    this.operator.lockedAddresses.forEach((stacker) => {
```

Comment:
Could something like this work?

*(Typed on the fly)*

```ts
sameDelegatedPoxAddrAllStackers =
  !this.operator.lockedAddresses.some(stacker =>
    model.wallets.get(stacker)!.delegatedPoxAddress !== firstDelegatedPoxAddress
  );
```
?

Reply:
And then `sameDelegatedPoxAddrAllStackers` can be `const`-bound.

Reply:
Second this^, the popular use-case is to "override" or "top-off" the same pool / index to include as many member as possible.

Reply:
@friedger @setzeus 
So a more suitable flow is:

- pick all the stackers' `delegated pox addresses`
- call `stack-aggregation-commit` for **all** the pox addresses found (e.g. found 3 addresses, call `stack-aggregation-commit` 3 times, one for each pox address)

Is this correct? If not, please let me know what other flow will suit the command better.

Reply:
Sounds good. You could also add a random value whether the pox address belongs to the pool or not.
Then you would have amybe only 2 commit txs

Reply:
Sounds good. You could also add a random value whether the pox address belongs to the pool or not.
Then you would have maybe only 2 commit txs